### PR TITLE
Additional refactoring for SQL Engine integration

### PR DESCRIPTION
**Pull request by fernst**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java [hunk 1]
```diff
@@ -50,10 +50,10 @@
  * The logic in this class has been extracted from the {@link AbstractBigQuerySink} in order to make this functionality
  * available to other classes in this package.
  */
-final class BigQuerySinkUtils {
+public final class BigQuerySinkUtils {
 
-  private static final String gcsPathFormat = "gs://%s/%s";
-  private static final String temporaryBucketFormat = gcsPathFormat + "/input/%s-%s";
+  public static final String GS_PATH_FORMAT = "gs://%s/%s";
+  private static final String temporaryBucketFormat = GS_PATH_FORMAT + "/input/%s-%s";
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java]: Could you please change this name to uppercase as well.

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java [hunk 2]
```diff
@@ -130,27 +147,42 @@ private static void createBucket(Storage storage, String bucket, @Nullable Strin
     }
   }
 
+  /**
+   * Updates {@link Configuration} with bucket details.
+   * Uses provided bucket, otherwise generates random.
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java]: It's actually not plain random, it's equal to runid.

**Review by tivv — COMMENTED** [review]
Could you also add some unit tests for this?

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java (hunk 2 above)
Comment (fernst) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java]: Good point. let me fix this.

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java (hunk 1 above)
Comment (fernst) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java]: Done.

**Review by tivv — COMMENTED** [review]
Please add a unit test for the utility method introduced.

**Comment by fernst** [comment]
@tivv Added a test for the `configureBucket` method.

**Review by tivv — APPROVED** [review]

**Pull request merged**